core/services/im_here/TrafficController.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python
+
+import asyncio
+from enum import Enum
+from typing import Dict, List, Tuple, Union
+
+import pynmea2
+from commonwealth.mavlink_comm.MavlinkComm import MavlinkMessenger
+from loguru import logger
+from pydantic import BaseModel, conint
+
+from exceptions import UnsupportedSocketKind
+from MavlinkNMEA import MavlinkGeoPackage, parse_mavlink_from_sentence
+
+
+class SocketKind(str, Enum):
+    """Available server sockets"""
+
+    UDP = "UDP"
+    TCP = "TCP"
+
+
+class NMEASocketInfo(BaseModel):
+    """Data representation of a NMEA server socket."""
+
+    kind: SocketKind
+    port: conint(gt=1023, lt=65536)  # type: ignore
```

Comment:
Can you explain what is happening here ?

Reply:
This is the serializable model for the NMEA socket. It contains the necessary information (network and mavlink-wise) to start a new sever socket.

Changed the name to just NMEASocket and refactored the docstrings to be more clear.